app/services/update_activity_as_programme.rb

```diff
@@ -1,20 +1,21 @@
-class CreateProgrammeActivity
-  attr_accessor :organisation_id, :fund_id
+class UpdateActivityAsProgramme
+  attr_accessor :activity, :parent_id
 
-  def initialize(organisation_id:, fund_id:)
-    self.organisation_id = organisation_id
-    self.fund_id = fund_id
+  def initialize(activity:, parent_id:)
+    self.activity = activity
+    self.parent_id = parent_id
   end
 
   def call
-    activity = Activity.new
-    activity.organisation = Organisation.find(organisation_id)
-    activity.reporting_organisation = activity.organisation
+    activity.extending_organisation = service_owner
 
-    fund = Activity.find(fund_id)
-    fund.child_activities << activity
+    activity.parent = begin
+                        Activity.fund.find(parent_id)
```

Comment:
I'm interested in the decision to guard against not finding the parent activity here as well, what led you to make that?